openshift_day2_configurator/configurators/ingress.py

```diff
@@ -0,0 +1,377 @@
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography import x509
+from cryptography.x509.oid import NameOID
+from datetime import datetime, timedelta
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+import os
+import base64
+import logging
+from typing import Any, Dict, Union, Optional
+from kubernetes.dynamic import DynamicClient
+
+from ocp_resources.resource import ResourceEditor
+from ocp_resources.configmap import ConfigMap
+from ocp_resources.secret import Secret
+from ocp_resources.pod import Pod
+from ocp_resources.ingress_controller import IngressController
+from ocp_resources.proxy import Proxy
+
+from openshift_day2_configurator.utils.general import (
+    verify_and_execute_configurator,
+)
+from rich.progress import Progress, TaskID
+from openshift_day2_configurator.utils.resources import create_ocp_resource
+
+CREATE_NEW_INGRESS_CERTIFICATE: str = "Create new Ingress certificate"
+CREATE_INGRESS_CERTIFICATE_CONFIGMAP: str = "Create ConfigMap with the new Ingress certificate"
+UPDATE_CLUSTER_PROXY_TRUSTED_CA: str = "Update cluster Proxy with the new Ingress certificate"
+CREATE_WILDCARD_CERTIFICATE_TLS_SECRET: str = "Create wildcard certificate tls secret"
+UPDATE_INGRESS_CONTROLLER_CERTIFICATE: str = "Update the IngressController certificate"
+WAIT_ON_INGRESS_PODS_RESCHEDULE: str = "Wait on Ingress pods to reschedule"
+
+OPENSHIFT_INGRESS_NAMESPACE: str = "openshift-ingress"
+INGRESS_CERTIFICATE_CONFIGMAP_NAME: str = "custom-ca"
+
+TIMEOUT_30MIN: int = 30 * 60
+
+
+def get_ingress_certificate_from_file(cluster_domain: str) -> str:
+    with open(f"*.{cluster_domain}.crt", "rb") as ingress_certificate_file:
+        ingress_certificate = ingress_certificate_file.read().decode("utf-8")
+
+    return ingress_certificate
+
+
+def wait_for_ingress_pod_ready_condition(ingress_pod: Pod) -> None:
+    ingress_pod.wait_for_condition(
+        condition=Pod.Condition.READY,
+        status=Pod.Condition.Status.TRUE,
+        timeout=TIMEOUT_30MIN,
+    )
+
+
+def create_new_ingress_certificate(
+    logger: logging.Logger,
+    cluster_domain: str,
+    ca_pem_file_path: str,
+    ca_key_file_path: str,
+) -> Dict[str, Dict[str, Union[str, bool]]]:
+    logger.debug(CREATE_NEW_INGRESS_CERTIFICATE)
+
+    missing_ca_files = [
+        ca_file_path for ca_file_path in [ca_pem_file_path, ca_key_file_path] if not os.path.exists(ca_file_path)
+    ]
+    if missing_ca_files:
+        logger.error(f"The following CA files does not exist: {missing_ca_files}")
```

Comment:
should fail here and not continue with this task

Reply:
done; added assertion